CONTRIBUTING.md

```diff
@@ -0,0 +1,18 @@
+### General checklist
+
+* Make sure your code is compliant to Coding guidelines section below.
+* Git: Use `git add -p` to stage relevant parts of a changed file, instead of the entire file. This will help you club related code in a commit despite them being split across files. (*and* you can work also concurrently work on other code changes).
```

Comment:
I'm not sure we should be dictating process to contributors, just expectations. It's better to demand that "a commit should only include the changes that are relevant to the issue being addressed in the commit."